src/mir/optimizations.md

```diff
@@ -26,6 +26,48 @@ optimizes it, and returns the improved MIR.
 [defid]: https://rustc-dev-guide.rust-lang.org/appendix/glossary.html#def-id
 [steal]: https://rustc-dev-guide.rust-lang.org/mir/passes.html?highlight=steal#stealing
 
+## Quickstart for adding a new optimization
+
+1. Make a Rust source file in `src/test/mir-opt` that shows the code you want to
+   optimize. This should be kept simple, so avoid `println!` or other formatting
+   code if it's not necessary for the optimization. The reason for this is that
+   `println!`, `format!`, etc. generate a lot of MIR that can make it harder to
+   understand what the optimization does to the test.
+
+2. Run `./x.py test --bless src/test/mir-opt/<your-test>.rs` to generate a MIR
+   dump. Read [this README][mir-opt-test-readme] for instructions on how to dump
+   things.
+
+3. Commit the current working directory state.
```

Comment:
Might be worth explaining a bit why you should commit this? I'm guessing so that you can see before/after?

Reply:
Yea, while you can see the diff, you'll still want to see the changes between your optimization and whatever was the previous state.